.clang-format

```diff
@@ -0,0 +1,151 @@
+## This config file is only relevant for clang-format version 8.0.0
```

Comment:
This should be a `STYLE:` commit, not a `WIP:`.

Reply:
Also, you could break it up into "adding style file" and "applying clang-format".

Reply:
I forgot to update that.  Will do.

Reply:
> Also, you could break it up into "adding style file" and "applying clang-format".

Will do.